addons/web/static/src/js/views/basic/basic_renderer.js

```diff
@@ -215,9 +215,9 @@ var BasicRenderer = AbstractRenderer.extend({
      * @param {integer} currentIndex
      * @return {integer}
      */
-    _activateNextFieldWidget: function (record, currentIndex) {
+    _activateNextFieldWidget: function (record, currentIndex, event) {
         currentIndex = (currentIndex + 1) % (this.allFieldWidgets[record.id] || []).length;
-        return this._activateFieldWidget(record, currentIndex, {inc: 1});
+        return this._activateFieldWidget(record, currentIndex, {inc: 1, event: event});
```

Comment:
I don't understand why this function needs an extra argument... I guess we always come from previous widget here

Reply:
Also, can't you use the 'inc' option instead?

Reply:
The event object is the one that holds information about what element triggered what event. `inc` isn't enough when you develop a widget such as the one in the description, where we have a matrix table and, i.e., hitting <kbd>TAB</kbd> in the last input of row 1 should lead the user to the 1st input on row 2, and the opposite with <kbd>Shift+TAB</kbd>.